### PR TITLE
Enable migration from previous pipeline versions

### PR DESCRIPTION
**Pull request by ajbozarth**
### What changes were proposed in this pull request?
Migration was moved to pipeline-services but not hooked back into the editor, this reimplements that functionality

Required upstream PR in pipeline-editor: https://github.com/elyra-ai/pipeline-editor/pull/140

Fixes https://github.com/elyra-ai/elyra/issues/1783

 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

**Comment by ajbozarth** [comment]
Some functionality is still broken:
- migrating from v0 to v1 throws an error in the console and fails to migrate
- migrate dialog will often open more than once (migration still works as intended, just a confusing UX)
- migration doesn't save and closing a migrated pipeline doesn't prompt for save on close

**Comment by lresende** [comment]
You will need to also update https://github.com/elyra-ai/elyra/blob/30290ad11794e4a4bf205515596b03de4d741176/elyra/cli/pipeline_app.py#L34

**Comment by ajbozarth** [comment]
Pushed fixes for:
- migrating from v0 to v1 throws an error in the console and fails to migrate
- migration doesn't save and closing a migrated pipeline doesn't prompt for save on close

Still need to address
- migrate dialog will often open more than once (migration still works as intended, just a confusing UX)

And the sister PR needs its tests updated.

File: Makefile [hunk 1]
```diff
@@ -85,10 +85,10 @@ lint-server: test-dependencies
 	flake8 elyra
 
 prettier-ui:
-	yarn prettier:check
+	yarn prettier
 
 eslint-ui:
-	yarn eslint:check --max-warnings=0
+	yarn eslint
```
Comment (bourdakos1) [inline Makefile:91]: is there a reason we are changing this?

File: packages/pipeline-editor/src/PipelineEditorWidget.tsx [hunk 2]
```diff
@@ -332,15 +336,55 @@ const PipelineWrapper: React.FC<IProps> = ({
...
+          console.log('migrating pipeline');
+          setPipeline(
+            migrate(pipeline, pipeline =>
+              PipelineService.setNodePathsRelativeToPipeline(
+                pipeline,
+                contextRef.current.path
+              )
+            )
+          );
+          contextRef.current.model.dirty = true;
```
Comment (bourdakos1) [inline packages/pipeline-editor/src/PipelineEditorWidget.tsx:370]: You should avoid calling `setPipeline` directly. We have the react state of the pipeline connected to file changes so using `contextRef.current.model.fromString("<pipeline-string>")` will handle the dirty state and the react pipeline state

**Review by bourdakos1 — CHANGES_REQUESTED** [review]

**Comment by kevin-bates** [comment]
Heads up. The work taking place in #1882 will require some changes as well.  To summarize those changes, parameters specific to the component (node) will be pushed down a level (regardless of component type) into a stanza named (TBD) `component_parameters`.  Only system-level/non-component properties will reside as siblings to `component_parameters`.  

For those pipelines prior to 3.0, we will need to identify at least one of the following: properties that should be located in the new `component_parameters` stanza, or system-level properties that should **not** be located in the new stanza.  I personally suspect the latter set might be easier and more stable when taking into account possible hotfixes that could hypothetically be applied, thereby preventing updates to migration code.  But that's purely hypothetical.

**Comment by kevin-bates** [comment]
Since all of the pipelines in https://github.com/elyra-ai/elyra/tree/master/elyra/pipeline/tests/resources/sample_pipelines will require updates to complete #1882, and those updates are exactly what should result from migration, it would be awesome if we could apply the migration to those prior to completion of #1882 and commit those updated files as part of the #1882 PR.  This would imply a slight window between this PR and #1882 in which a migrated pipeline would be unusable until #1882 is merged, but I believe that could be coordinated rather easily and we can benefit in a couple of ways - practical migration testing and time/stability savings of manual updates.

File: Makefile (hunk 1 above)
Comment (ajbozarth) [reply Makefile:91]: yeah, I'm unsure why this was changed, I would've not approved it myself, it removes functionality that was intentional. Changing this to check just makes lint fail with no help to the user. Prettier is meant to fix and the fix flag on eslint only make minor intentional changes, the rest are still just displayed, though on a re-read I should leave `--max-warnings=0` as that is useful to add. The use of fix here matches how jupyterlab lint functions

File: packages/pipeline-editor/src/PipelineEditorWidget.tsx (hunk 2 above)
Comment (ajbozarth) [reply packages/pipeline-editor/src/PipelineEditorWidget.tsx:370]: gotcha, I'll switch to using that

File: Makefile (hunk 1 above)
Comment (bourdakos1) [reply Makefile:91]: Maybe we need to distinguish between what's run on ci vs locally? I don't think that eslint/prettier should stop a local build. On ci we want it to fail, if changes need to be made to the code

**Comment by ajbozarth** [comment]
Based on Webex convo with @kevin-bates I will be removing the update to v4 from this PR and we will make that switch alongside writing the migration in  #1882 where the actual changes that need migration are added.

**Comment by kevin-bates** [comment]
Related to this https://github.com/elyra-ai/elyra/pull/1860#issuecomment-877290570, @kiersten-stokes and I decided it would be best to manually update the server-based test pipeline files now (in this [commit](https://github.com/elyra-ai/elyra/pull/1882/commits/574311cebbabb1233682818c78c26d2b24968c8a)) since not having them updated was interfering with our ability to validate the actual changes for #1882.

File: Makefile [hunk 3]
```diff
@@ -85,10 +85,10 @@ lint-server: test-dependencies
 	flake8 elyra
 
 prettier-ui:
-	yarn prettier:check
+	yarn prettier
 
 eslint-ui:
-	yarn eslint:check --max-warnings=0
+	yarn eslint --max-warnings=0
```
Comment (bourdakos1) [inline Makefile:91]: can you rebase? this should be fixed in master now

**Review by bourdakos1 — APPROVED** [review]

File: Makefile (hunk 3 above)
Comment (ajbozarth) [reply Makefile]: done

**Comment by marthacryan** [comment]
I'm using an old pipeline from the https://github.com/CODAIT/covid-notebooks repo and I'm seeing the dialog show up twice: 
Jul-13-2021 14-35-41

**Comment by marthacryan** [comment]
Oh oops didn't see that that was a known issue ^^ I'm fine with that being a follow-up PR.

**Review by marthacryan — APPROVED** [review]
Tried locally and other than the issue with the dialog showing up more than once, this looks good

**Review by lresende — APPROVED** [review]

**Pull request merged**